provisioning/device/src/CertificateInstaller.cs

```diff
@@ -1,68 +1,31 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.Azure.Devices.Shared;
-using System;
-using System.Collections.Generic;
 using System.Security.Cryptography.X509Certificates;
+using Microsoft.Azure.Devices.Shared;
 
 namespace Microsoft.Azure.Devices.Provisioning.Client
 {
     internal static class CertificateInstaller
     {
-        private static readonly HashSet<string> s_installedCertificates = new HashSet<string>();
-        private static readonly object s_lock = new object();
-
-        static CertificateInstaller()
-        {
-            try
-            {
-                using (var store = new X509Store(StoreName.CertificateAuthority, StoreLocation.CurrentUser))
-                {
-                    store.Open(OpenFlags.ReadOnly);
-                    foreach (X509Certificate2 certificate in store.Certificates)
-                    {
-                        s_installedCertificates.Add(certificate.Thumbprint);
-                    }
-                }
-            }
-            catch (Exception ex)
```

Comment:
The exception handling is important: if certs fail to install you would want to get the information in logs. 
It could mean that either the process can't access the stores (device won't authenticate) or that some corruption happened with the store. I remember seeing the last one in SslStream testing (quite rare but very hard to debug).